DiscordBot/moderator.py

```diff
@@ -0,0 +1,260 @@
+from enum import Enum, auto
+import discord
+import json
+import re
+from supabase import create_client, Client
+
+with open("tokens.json", "r") as f:
+    tokens = json.load(f)
+
+supabase_url = tokens.get("SUPABASE_URL")
+supabase_key = tokens.get("SUPABASE_KEY")
+
+supabase: Client = create_client(supabase_url, supabase_key)
+
+class ModeratorState(Enum):
+    EVAL_START = auto()
+    AWAITING_VERACITY_DECISION = auto()
+    AWAITING_LEGIT_REPORT_DECISION = auto()
+    AWAITING_FALSE_REPORT_DECISION = auto()
+    AWAITING_REPORT_FREQUENCY_DECISION = auto()
+    AWAITING_HIDE_CHILDREN_DECISION = auto()
+    AWAITING_BAN_DECISION = auto()
+    AWAITING_DECISION = auto()
+    AWAITING_ACTION = auto()
+    ACTION_COMPLETE = auto()
+
+class ModeratorReport:
+    START_KEYWORD = "eval"
+    CANCEL_KEYWORD = "cancel"
+    HELP_KEYWORD = "help"
+
+    YES_NO_OPTIONS = [
+        "Yes",
+        "No"
+    ]
+    VERACITY_OPTIONS = [
+        "Yes",
+        "No",
+        "Need more information. Escalate to next moderator team"
+    ]
+    LEGIT_REPORT_DECISION_OPTIONS = [
+        "Escalate to law enforcement",
+        "It does not violate policy/the law, but not appropriate behavior (e.g., text a minor saying they are cute)"
+    ]
+    FALSE_REPORT_DECISION_OPTIONS = [
+        "Warn user about not submitting false reports",
+        "Increase the number of times user has submitted false reports"
+    ]
+    REPORT_FREQUENCY_DECISION_OPTIONS = [
+        "Report to NCMEC",
```

Comment:
Should we just change this to law enforcement and prompt the user to report to both NCMEC and Lantern?

Something like:

"Please report to NCMEC here and Lantern here. When you are done with the reporting, type Yes"

Reply:
By user do you mean the mod going through the mod flow or the user who made the report? I think if its still going to be up to the mod to report to NCMEC and Lantern we should just leave it as an option and assume that by picking the option the reports are made by the system

Reply:
Makes sense!